websites/S/Shiro/presence.ts

```diff
@@ -0,0 +1,294 @@
+      }
+    },
+    "/anime/"(){
+      PageStatus = "viewAnime";
+
+      const title = document.querySelector('h1 > div')?.textContent || "Loading...";
+
+      presenceData.smallImageKey = "search";
+      presenceData.startTimestamp = startTimestamp;
+
+      presenceData.buttons = [{
+        label: "View Anime",
+        url: document.baseURI
+      }];
+
+      presenceData.startTimestamp = startTimestamp;
+
+      return { title };
+    },
+    "/browse/"(){
+      const SearchQuery = (new URLSearchParams(document.location.search)).get("search");
+
+      if (SearchQuery){
+        PageStatus = "Searching";
+
+        presenceData.details = "Seaching for:";
+        presenceData.state = SearchQuery;
+
+        presenceData.smallImageKey = "search";
+        presenceData.startTimestamp = startTimestamp;
+
+        return { SearchQuery };
+      } else {
+        presenceData.details = "Browsing...";
+        presenceData.startTimestamp = startTimestamp;
+      }
+    }
+  };
+
+  for (const [pathname, setPresenceData] of Object.entries(pages)){
+    if (document.location.pathname.includes(pathname)){
+      if (!PageStatus) PageStatus = pathname.replace(/\//g, "");
+      pagesData[PageStatus] = setPresenceData();
+    }
+  }
+
+  for (const [setting, settingData] of Object.entries(presenceSettings)){
+    if (typeof settingData !== "boolean"){
+      for (const condition of settingData.condition){
+        if (condition.page === PageStatus && condition.setTo && !condition.enabled && !condition.replace && !condition.delete){
+          presenceData[setting as "state" | "details"] = condition.setTo;
+
+        } else if (condition.page === PageStatus && !condition.enabled && condition.delete){
+          delete presenceData[setting as keyof PresenceData];
+
+        } else if (condition.page === PageStatus && condition.replace) {
+          let replaced = condition.setTo;
+
+          for (const replace of condition.replace.toRepalce) 
+          replaced = replaced.replace(replace.text, pagesData[condition.page][replace.with]);
+
+          presenceData[setting as "state" | "details"] = replaced;
+        } else if (condition.delete && !condition.enabled) delete presenceData[setting as keyof PresenceData];
+      }
+    } else if (!settingData) delete presenceData[setting as keyof PresenceData];
+  }
+
+  if (!presenceData.details){
+    presence.setActivity();
+    presence.setTrayTitle();
+  } else {
+    presence.setActivity(presenceData);
+  }
+});
+
+/**
+ * Check if you can see this Element
+ * @param element The element you want to check
+ * @returns The result you want
+ */
+
+function YouCanSeeThis(element: HTMLElement) {
+  const clientRect = element.getBoundingClientRect();
+  return (
+    clientRect.top >= 0 &&
+    clientRect.left >= 0 &&
+    clientRect.bottom <= (window.innerHeight || document.documentElement.clientHeight) &&
+    clientRect.right <= (window.innerWidth || document.documentElement.clientWidth)
+  );
+}
```

Comment:
For what purpose, fr
```typescript
function isHidden(element: HTMLElement) {
    return (!element.offsetParent)
}
``` 
is easier and it has less code.

Reply:
That isn't what I want, what I want is just to check if that Element if visble on sceen or not

Reply:
Literally, unnecessary for me but FINE.

Reply:
;-;